### ISSUE TITLE
High disk usage with nested docs

### ISSUE THREAD

**Issue by abulhol** [elastic/elasticsearch#16721]
I have posted this question on the ES forum a while ago:
https://discuss.elastic.co/t/nesting-high-disk-usage/40626
What makes my index use such high amounts of disk space, when it consumes so little without nesting?

**Comment by ebuildy** [comment]
Hello,

Nested data type creates a separate document when indexing, whereas object will result as flattens object hierarchies into a simple list of field names and values

As you can read on the doc: 

https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html#_using_literal_nested_literal_fields_for_arrays_of_objects

So there is some overhead because document has an uuid, version fields etc..

**Comment by abulhol** [comment]
Hi Thomas,
Thank you for the explanation. I have made a calculation:
I have indexed 615 docs, once with and once without nesting.
With nesting, I have 17,123,628 docs and 15.4 GB.
Without nesting, it is 615 docs and 1.6 GB.
(using curl -XGET '127.0.0.1:9200/_my_index_/_stats/store,docs?pretty&human')

So the overhead per doc seems to be approximately (15.4GB-1.6GB)/(17123628-615) = 0.85 KByte

Is that what you would expect as well?
